app/pages/project/classify.jsx

```diff
@@ -128,34 +82,45 @@ export class ProjectClassifyPage extends React.Component {
     }
   }
 
-  shouldWorkflowAssignmentPrompt(nextProps) {
-    // Only for Gravity Spy which is assigning workflows to logged in users
-    if (nextProps.project.experimental_tools.indexOf('workflow assignment') > -1) {
-      const assignedWorkflowID = nextProps.preferences && nextProps.preferences.settings && nextProps.preferences.settings.workflow_id;
-      const currentWorkflowID = this.props.preferences && this.props.preferences.preferences.selected_workflow;
-      if (assignedWorkflowID && currentWorkflowID && assignedWorkflowID !== currentWorkflowID) {
-        if (this.state.promptWorkflowAssignmentDialog === false) {
-          this.setState({ promptWorkflowAssignmentDialog: true });
+  componentWillUpdate(nextProps) {
+    const nextWorkflowID = (isPresent(nextProps) && isPresent(nextProps.workflow)) ? nextProps.workflow : null;
+    this.context.geordi.remember({ workflowID: nextWorkflowID });
+  }
+
+  componentDidUpdate(prevProps) {
+    const { actions, currentClassifications, project, upcomingSubjects, workflow } = this.props;
+
+    if (project !== prevProps.project) {
+      this.loadAppropriateClassification();
```

Comment:
Classifications are keyed by workflow ID so why does the classifier load a new classification when the project changes?